Lib/Src/CUartCom.cpp

```diff
@@ -98,25 +100,40 @@ void CUartCom::stopRx()
 }
 
 /**
- * @brief Add data to the TX Queue and start transmission if possible
- * @param msg Message to send.
- * @return True if message was added to the queue and transmission started
- * successfully.
+ * @brief send string
+ * @param msg Message to be sent via UART
+ * @return success if message was successfully stored in TX Queue.
  */
 bool CUartCom::send(etl::string<MAX_STRING_SIZE> msg)
 {
-    bool b_success = false;
-    // Add message to queue
-    if (m_tx_queue.size() <= MAX_TX_QUEUE_SIZE && (msg.empty() == false))
+    uint32_t msg_len = msg.length();
+    bool b_success = send((uint8_t *)msg.c_str(), msg_len);
+    return b_success;
+}
+
+/**
+ * @brief Add data to the TX Queue and start transmission if possible
+ * @param p_data_buf Reference to array of uint8_t data
+ * @param len Size of data that needs to be stored
+ * @return True if data was added to the queue, false in case of buffer overflow
+ */
+bool CUartCom::send(uint8_t *p_data_buf, uint32_t len)
+{
+    bool b_success = true;
+
+    // Store data in tx_queue
+    for (uint32_t i = 0; i < len; i++)
     {
-        m_tx_queue.put(msg);
-        b_success = true;
+        if (m_tx_queue.put(p_data_buf[i]) == false)
+        {
+            b_success = false;
+        }
     }
 
-    // Start transmission only if UART is idle
```

Comment:
haha. just realised you had a better comment before, but changed it for some reason.